### ISSUE TITLE
Workload publicEndpoints annotation isn't updated on edit

### ISSUE THREAD

**Issue by mantis-toboggan-md** [rancher/dashboard#4747]
This report spawned from this issue: https://github.com/rancher/dashboard/issues/4398#issuecomment-982337011 -- keeping this ticket in dashboard since there is still UI work to be done.

> ## Validation steps
> ### ❌ Scenario 1 | Node Port | Blank Name | Change Listening Port

> The `Endpoints` URL is not being updated on the `Deployments` page. Even though it's actually being changed if you navigate to the correct URL. See the screen shot and video listed below. Screen shot labeled `A1`

> # My checks FAILED
> ## Validation Environment
> Component	Version / Type
> Rancher version	v2.6-4375161a2780765f0fd58af2936876a3e0254741-head
> Rancher cluster type	Docker
> Docker version	20.10.7, build 20.10.7-0ubuntu5~20.04.2
> Helm version	v2.16.8-rancher1
> Downstream cluster type	RKE1 Linode
> Downstream K8s version	v1.21.6-rancher1-1
> Browser type	Google Chrome
> Browser version	96.0.4664.55 (Official Build) (x86_64)
> ## Validation steps
> ### ❌ Scenario 1 | Node Port | Blank Name | Change Listening Port
> ### TLDR:
> 1. Starting from the **default** Rancher homepage `/dashboard/home`
> 2. Click `hamburger menu` -> click `Cluster Management` -> click `Explore` (next to cluster) -> click `Workload` -> click `Deployments` -> click `Create`
> 3. Fill out the deployment form
>    
>    * Namespace: `default`
>    * Name: `hello-world`
>    * Container Image: `brudnak/hello-rancher-golang`
>    * Click `Add Port`
>      
>      * Service Type: `Node Port`
>      * Name: leave **name** field blank
>      * Private Container Port: `3000`
>      * Listening Port: `30441`
>    * Click `Create` -> to create the deployment
> 4. Deployment successfully working
> 
> > User decides they want to change the listening port from **30441** to **30442**
> 
> 5. Click the `ellipsis` next to the `hello-world` deployment
> 6. Click `Edit Config`
> 7. Observe the port section while editing the config, the form isn't picking up previous selections
>    
>    * Service Type:
>      
>      * Was: `Node Port`
>      * Now: `Do not create a service`
> 8. Change service type back to `Node Port`
> 9. Leave `Name` as is, it should be a generated name
> 10. Enter newly desired Listening Port `30442`
> 
> **----------------------------------------------------------------------------------------------**
> 
> **INITIAL RESULT**
> 
> You are redirected back to the deployments page. The hello-world deployment remains active, the `Endpoints` link does not get updated. It still showing `30441/TCP`. If you click this link you get page not found, however if you manually change the URL to port `30442` you can see the deployment.
> 
> **----------------------------------------------------------------------------------------------**
> 
> 11. Edit the config again as in steps `6 through 10` and change the listening port to `30443`.
> 12. You'll see the same results, where the UI is still showing `30441/TCP` for the `Endpoints` link. Even though we've changed it twice now to `30442` and now `30443`. If you change the URL to port
> 
> **RESULT**
> 
> The `Endpoints` URL is not being updated on the `Deployments` page. Even though it's actually being changed if you navigate to the correct URL. See the screen shot and video listed below. Screen shot labeled `A1`
> 
> **SCREENSHOT**
> 
> ![Screen Shot 2021-11-29 at 11 28 52 AM](https://user-images.githubusercontent.com/40877149/143999372-74bb56e2-0095-4c9e-b2b1-cfe263e583eb.png)
> 
> **VIDEO**
> 
>  Screen.Recording.2021-11-29.at.3.11.43.PM.mov 

**Additional context**
I've investigated the UI side of the scenario above and it looks like a backend issue. The 'Endpoints' column in the workload table is using the `field.cattle.io/publicEndpoints` annotation and that isn't updating when the workload/associated services change.

**Comment by jordojordo** [comment]
#4744

**Comment by brudnak** [comment]
# My checks PASSED
 
## Reproduction Environment


#### Not required.

<hr>
  
## Validation Environment

| Component               | Version / Type                                     |
| ----------------------- | -------------------------------------------------- |
| Rancher version         | v2.6-fd930f76545c64fdad8b685d86090a8fbb8bebcf-head |
| Rancher cluster type    | Docker                                             |
| Docker version          | 20.10.7, build 20.10.7-0ubuntu5~20.04.2            |
| Helm version            | v2.16.8-rancher1                                   |
| Downstream cluster type | RKE1 Linode                                        |
| Downstream K8s version  | v1.21.7-rancher1-1                                 |
| Browser type            | Opera GX                                           |
| Browser version         | LVL3 (core: 82.0.4227.25) (x86_64)                 |

## Validation steps

1. Create deployment
    - Image used: `brudnak/hello-rancher-golang`
    - Node Port Details
        - Private Container Port: `3000`
        - Listening Port: `30441`
        - Name: No name provided
2. Edit the deployment and change the listening port from `30441` to `30442`

**RESULT**

Getting the expected behavior of the UI updating the current listening port.
